File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.cpp
#include <bits/stdc++.h>

using namespace std;

bool cmp (string a, string b) {
    string x = a + b;
    string y = b + a;
    if(stoi(x) > stoi(y)) return true;
    else return false;
}

string solution(vector<int> numbers) {
    string answer = "";
    
    vector<string> strings;
    for(auto number : numbers) {
        strings.push_back(to_string(number));
    }
    
    sort(strings.begin(), strings.end(), cmp);
    if(strings[0][0] == '0') {
        answer = "0";
    }
    else{ 
        for(auto str : strings) {
            answer += str;
        }   
    }
    
    return answer;
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key

def compare(a, b):
    return -(int(a+b) - int(b+a)) 

def solution(numbers):
    numbers = list(map(str, numbers))
    numbers.sort(key=cmp_to_key(compare))
    if numbers[0] == "0":
        return "0"
    return ''.join(numbers)